parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -227,6 +239,45 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return internalReader.nextKeyValue();
   }
 
+  /**
+   * Reads the next batch of rows. This method is used for reading primitive types
+   * and does not call the converters at all.
+   * @param previous a row batch object to be reused by the reader if possible
+   * @return the row batch that was read
+   * @throws java.io.IOException
+   */
+  public RowBatch nextBatch(RowBatch previous) throws IOException {
+    MessageType requestedSchema = readContext.getRequestedSchema();
+    List<ColumnDescriptor> columns = requestedSchema.getColumns();
+    int nColumns = columns.size();
+    ColumnVector[] columnVectors;
+
+    RowBatch rowBatch = previous;
+    if (rowBatch == null) {
+      rowBatch = new RowBatch(new ColumnVector[nColumns]);
+    }
+
+    columnVectors = rowBatch.getColumns();
+    MessageType[] columnSchemas = new MessageType[nColumns];
+    for (int i = 0; i < nColumns; i++) {
+      ColumnVector columnVector = columnVectors[i];
```

Comment:
If RowBatch 'previous' is passed as non-null and with less number of columns than 'requestedSchema', then I think this line will throw an ArrayIndexOutOfBoundException. Being this a public method, we should validate that both 'previous' and 'requestedSchema' have the same number of columns and throw a better error message otherwise.